### DISCUSSION TITLE
Make libusb compulsory?

### DISCUSSION THREAD

**Discussion by hughsie** [fwupd/fwupd#7411] [Ideas]
I've got a branch that drops GUsb as a dep and instead uses libusb directly. I've not started sprinkling `#ifdef LIBUSB` everywhere, and I'm wondering if we can just make it a hard dep. The only target we have without USB is openbmc, and I'm wondering if that's worth all the extra complexity.

**Comment by superm1** [comment]
Is it really a lot of `#ifdef LIBUSB`?  I would think it's not much different than disabling the meson option for gusb.  Just new meson option for libusb and swap those over?

**Reply by hughsie** [reply]
> Is it really a lot of #ifdef LIBUSB?

It's 49 so far.

**Reply by superm1** [reply]
Yeesh; OK I see the point then.  I'm fine with making it mandatory, but you probably need to ask the openbmc people what they think if they're the only ones that use it what it does to their builds.

**Reply by hughsie** [reply]
I don't even think it's included in openbmc by default -- it was a PoC for a big search engine.

**Reply by hughsie** [reply]
Ohh -- I just realized why it's important. The fuzzing targets don't link libusb.

**Reply by hughsie** [reply]
Okay, fixed by only compiling the needed bits in the fuzzing targets. I think 15 ifdefs is a lot more sane.
